### PR TITLE
More game dialog fixes and add OSK message

### PR DESCRIPTION
**Pull request by Megamouse**
So far I fixed the Nier regression 🤦‍♂️.

Update: 
I hope all the games which worked before still work with this PR. After my initial facepalm commit I fixed some weird behaviours with Diablo 3 that ultimately led to the conclusion that the facepalm commit was wrong (There was a reason why It was like that after all).
Sadly I can't find a proper fix for Class of Heroes that doesn't crash Diablo 3, so I'll give up for now. It seems like there is only a small detail missing though, so maybe I'll find it sometime in the Nier future.

Update 2:
Finally added the message to the On-Screen-Keyboard. I have no clue why we forgot about that one.
image

### CODE REVIEW

**Comment by itinga2019** [comment]
Please, when will you merge this build?

**Comment by RainbowCookie32** [comment]
When it's ready @itinga2019

File: rpcs3/Emu/Cell/Modules/cellOskDialog.cpp [hunk 1]
```diff
@@ -20,13 +20,16 @@ s32 cellOskDialogLoadAsync(u32 container, vm::ptr<CellOskDialogParam> dialogPara
 
 	if (!inputFieldInfo || !inputFieldInfo->message || !inputFieldInfo->init_text || inputFieldInfo->limit_length > CELL_OSKDIALOG_STRING_SIZE)
 	{
+		cellOskDialog.error("cellOskDialogLoadAsync: CELL_OSKDIALOG_ERROR_PARAM");
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellOskDialog.cpp]: It's like `error_code` never existed.

Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellOskDialog.cpp]: It wouldn't compile

Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellOskDialog.cpp]: You were probably missing required string formatting for the error codes.

File: rpcs3/Emu/Cell/Modules/cellOskDialog.cpp [hunk 2]
```diff
@@ -23,59 +40,127 @@ s32 cellOskDialogLoadAsync(u32 container, vm::ptr<CellOskDialogParam> dialogPara
...
+						osk->osk_text[i] = string_to_send[i];
+					}
+
+					done = true;
+					return 0;
+				});
+
+				while (done == false)
+				{
+					// wait for check callback
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellOskDialog.cpp]: Dont busy wait here, use atleast a thread yield.

**Pull request merged**